blazor/diagram/interaction.md

```diff
@@ -89,7 +89,208 @@ Get the currently selected items from the [Nodes](https://help.syncfusion.com/cr
+        groupNode.ID = RandomId();
+        List<string> child = new List<string>();
+        foreach (string childID in groupNode.Children)
+        {
+            for (int i = 0; i < diagram.Nodes.Count; i++)
+            {
+                if (childID == diagram.Nodes[i].ID && diagram.Nodes[i] is NodeGroup nodeGroup)
+                {
+                    child.Add(CloneGroup(nodeGroup, true));
+                }
+                else if (childID == diagram.Nodes[i].ID)
+                {
+                    child.Add(CloneNode(diagram.Nodes[i], true));
+                }
+            }
+            for (int i = 0; i < diagram.Connectors.Count; i++)
+            {
+                if (childID == diagram.Connectors[i].ID)
+                {
+                    child.Add(CloneConnector(diagram.Connectors[i], true));
+                }
+            }
+        }
+        groupNode.Children = child.ToArray();
+        if (!isChild)
+        {
+            groupNode.OffsetX += 25;
+            groupNode.OffsetY += 25;
+        }
+        diagram.AddDiagramElements(new DiagramObjectCollection<NodeBase>() { groupNode });
+        return groupNode.ID;
+    }
+    public string CloneNode(Node node, bool isChild)
+    {
+        diagram.StartGroupAction();
+        Node nodeChild = node.Clone() as Node;
+        nodeChild.ID = RandomId();
+        if (!isChild)
+        {
+            nodeChild.OffsetX += 25;
+            nodeChild.OffsetY += 25;
+        }
+        diagram.AddDiagramElements(new DiagramObjectCollection<NodeBase>() { nodeChild });
+        diagram.EndGroupAction();
+        return nodeChild.ID;
+    }
+    public string CloneConnector(Connector connector, bool isChild)
+    {
+        diagram.StartGroupAction();
+        Connector connectorChild = connector.Clone() as Connector;
+        connectorChild.ID = RandomId();
+        if (!isChild)
+        {
+            connectorChild.SourcePoint = new DiagramPoint() { X = connectorChild.SourcePoint.X + 25, Y = connectorChild.SourcePoint.Y + 25 };
+            connectorChild.TargetPoint = new DiagramPoint() { X = connectorChild.TargetPoint.X + 25, Y = connectorChild.TargetPoint.Y + 25 };
+        }
+        diagram.AddDiagramElements(new DiagramObjectCollection<NodeBase>() { connectorChild });
+        diagram.EndGroupAction();
+        return connectorChild.ID;
+    }
 
+    internal string RandomId()
+    {
+        Random random = new Random();
+        const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXTZabcdefghiklmnopqrstuvwxyz";
+#pragma warning disable CA5394 // Do not use insecure randomness
+        return new string(Enumerable.Repeat(chars, 5)
+          .Select(s => s[random.Next(s.Length)]).ToArray());
+#pragma warning restore CA5394 // Do not use insecure randomness
+    }
+}
+```
+You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/Blazor-Diagram-Examples/tree/master/UG-Samples/Interaction)
```

Comment:
You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/Blazor-Diagram-Examples/tree/master/UG-Samples/Interaction).